scripts/process/time_filters.R

```diff
@@ -1,31 +1,61 @@
-process.month_filter <- function(viz = as.viz("month_filter")){
+process.time_filter <- function(viz = as.viz("month_filter")){
   library(dplyr)
-  
+
   viz.data <- readDepends(viz)[["year_filter"]]
-  
-  range_text <- "-1 month"
-  
+
+  range_text <- viz[["range_text"]]
+
   range_days = seq(max(viz.data$date, na.rm = TRUE), length = 2, by = range_text)
-  
+
   month_data <- viz.data %>%
-    filter(date >= range_days[2]) 
-  
+    filter(date >= range_days[2])
+
   saveRDS(month_data, file=viz[["location"]], compress = FALSE)
-  
+
 }
 
-process.week_filter <- function(viz = as.viz("week_filter")){
+process.fill_missing_year <- function(viz = as.viz("fill_missing_year")){
   library(dplyr)
-  
+  library(tidyr)
+
   viz.data <- readDepends(viz)[["year_filter"]]
-  
-  range_text <- "-1 week"
-  
+
+  max_date <- Sys.Date() - 1
+  range_text <- "-1 year"
+  year_days = seq(max_date, length = 2, by = range_text)
+
+  full_dates <- seq(max_date, length = -as.numeric(diff(year_days)), by=-1)
+
+  sub_data <- select(viz.data, date, sessions, newUsers, viewID) %>%
+    group_by(date, viewID) %>%
+    summarize(sessions = sum(sessions, na.rm = TRUE),
+              newUsers = sum(newUsers, na.rm = TRUE)) %>%
+    ungroup() %>%
+    bind_rows(data_frame(viewID=NA, date=full_dates)) %>%
+    unite(old_new, sessions,newUsers) %>%
```

Comment:
I wanted to spread and fill both the "sessions" and "newUsers" columns. It's briefly `unite`d into a column called "old_new"...but "all_new" would make more sense, yes, I'll change that to avoid future confusion